File: Navtrack.Web/Models/Locations/LocationHistoryRequestModel.cs
using System;

namespace Navtrack.Web.Models.Locations
{
    public class LocationHistoryRequestModel
    {
        public int AssetId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public decimal? Latitude { get; set; }
        public decimal? Longitude { get; set; }
        public int? Radius { get; set; }
        public int? StartSpeed { get; set; }
        public int? EndSpeed { get; set; }
        public int? StartAltitude { get; set; }
        public int? EndAltitude { get; set; }
    }
}

File: Navtrack.Core/ClientApp/src/apis/types/location/LocationHistoryRequestModel.ts
export type LocationHistoryRequestModel = {
    assetId: number;
    startDate: string;
    endDate: string;
    latitude?: number;
    longitude?: number;
    radius?: number;
    startSpeed?: number;
    endSpeed?: number;
    startAltitude?: number;
    endAltitude?: number;
};
